File: api/src/main/java/me/vihara/justahologram/api/hologram/Hologram.java
package me.vihara.justahologram.api.hologram;

import me.vihara.justahologram.api.hologram.line.HologramLine;
import me.vihara.justahologram.api.property.map.PropertyMap;

import java.util.HashSet;

public interface Hologram {
    String id();
    Position position();
    HashSet<HologramLine> lines();
    PropertyMap properties();

    class Position {
        public String worldName;
        public double x;
        public double y;
        public double z;
        public float yaw;
        public float pitch;

        public Position(String worldName, double x, double y, double z, float yaw, float pitch) {
            this.worldName = worldName;
            this.x = x;
            this.y = y;
            this.z = z;
            this.yaw = yaw;
            this.pitch = pitch;
        }
    }
}

File: plugin/src/main/kotlin/me/vihara/plugin/justahologram/hologram/Hologram.kt
package me.vihara.plugin.justahologram.hologram

import me.vihara.justahologram.api.hologram.IHologram
import me.vihara.justahologram.api.hologram.line.IHologramLine
import me.vihara.justahologram.api.property.map.PropertyMap

data class Hologram(val id: String, val position: IHologram.Position, val lines: HashSet<IHologramLine>, val properties: PropertyMap) :
    IHologram {
    var lastTick = 0

    override fun id(): String {
        return id
    }

    override fun position(): IHologram.Position {
        return position
    }

    override fun lines(): HashSet<IHologramLine> {
        return lines
    }

    override fun properties(): PropertyMap {
        return properties
    }
}